packages/ses/scripts/bundle.js

```diff
@@ -15,40 +17,59 @@ const write = async (target, content) => {
   await fs.promises.writeFile(location, content);
 };
 
-const main = async () => {
+/**
+ * @param {object} [options]
+ * @param {string} [options.buildType] Suffix used to build special bundles (e.g. 'hermes')
+ */
+const writeBundle = async ({ buildType } = {}) => {
   const text = await fs.promises.readFile(
     fileURLToPath(`${root}/package.json`),
     'utf8',
   );
   const packageJson = JSON.parse(text);
   const version = packageJson.version;
 
+  const moduleTransforms = {};
+
+  let bundleFilePaths = [
+    'dist/ses.cjs',
+    'dist/ses.mjs',
+    'dist/ses.umd.js',
+    'dist/lockdown.cjs',
+    'dist/lockdown.mjs',
+    'dist/lockdown.umd.js',
+  ];
+  let terseFilePaths = ['dist/ses.umd.min.js', 'dist/lockdown.umd.min.js'];
+
+  if (buildType === 'hermes') {
+    bundleFilePaths = ['dist/ses-hermes.cjs'];
+    terseFilePaths = [];
+    Object.assign(moduleTransforms, hermesTransforms);
+  }
+
   const bundle = await makeBundle(
     read,
     pathToFileURL(resolve('../index.js', import.meta.url)).toString(),
+    // @ts-expect-error TS2322 Type '{}' is not assignable to type 'ModuleTransforms'. Index signature for type 'string' is missing in type '{}'.
+    { moduleTransforms },
   );
   const versionedBundle = `// ses@${version}\n${bundle}`;
 
-  const { code: terse } = await minify(versionedBundle, {
-    mangle: false,
-    keep_classnames: true,
-  });
-  assert.string(terse);
-
+  console.log(`-- Building '${buildType}' version of SES --`);
   console.log(`Bundle size: ${versionedBundle.length} bytes`);
-  console.log(`Minified bundle size: ${terse.length} bytes`);
 
-  await fs.promises.mkdir('dist', { recursive: true });
+  let terse;
+  if (terseFilePaths.length) {
+    const { code } = await minify(versionedBundle, {
+      mangle: false,
+      keep_classnames: true,
+    });
+    terse = code;
+    assert.string(terse);
+    console.log(`Minified bundle size: ${terse.length} bytes`);
+  }
 
-  const bundleFilePaths = [
-    'dist/ses.cjs',
-    'dist/ses.mjs',
-    'dist/ses.umd.js',
-    'dist/lockdown.cjs',
-    'dist/lockdown.mjs',
-    'dist/lockdown.umd.js',
-  ];
-  const terseFilePaths = ['dist/ses.umd.min.js', 'dist/lockdown.umd.min.js'];
+  await fs.promises.mkdir('dist', { recursive: true });
 
   await Promise.all([
     ...bundleFilePaths.map(dest => write(dest, versionedBundle)),
```

Comment:
```suggestion
    ...bundleFilePaths.map(dest => write(dest, versionedBundle)),
    minify(versionedBundle, { mangle: false, keep_classnames: true }).then(({ code }) => {
      assert.string(code);
      console.log(`Minified bundle size: ${code.length} bytes`);
      return terseFilePaths.map(dest => write(dest, terse));
    });
```
(could) make the top write go ahead without waiting for minification

Reply:
this would replace lines 61-70 right

https://github.com/endojs/endo/blob/7e4f681ed36b7e9d211ba727dfd31d5f89659e68/packages/ses/scripts/bundle.js#L61-L70

Reply:
i've updated the suggestion to preserve the condition `terseFilePaths.length &&` to avoid minifying on Hermes
(then updated `write(dest, terse)` to `write(dest, code)` to avoid the rename dance)
(finished with a `,` instead of `;` as the 2nd/last Promise.all array item)

```suggestion
    ...bundleFilePaths.map(dest => write(dest, versionedBundle)),
    terseFilePaths.length &&
      minify(versionedBundle, {
        mangle: false,
        keep_classnames: true,
      }).then(({ code }) => {
        assert.string(code);
        console.log(`Minified bundle size: ${code.length} bytes`);
        return terseFilePaths.map(dest => write(dest, code));
      }),
```

but i think looks a bit messy, perhaps worth keeping it split into the `if` statement like it was before?

Reply:
Agreed that inlining the `terseFilePaths.length` conditional there is not great for readability... Maybe it's not worth it like you say. Let me just propose one alternative for consideration :)

```suggestion
    ...bundleFilePaths.map(dest => write(dest, versionedBundle)),
    ...(terseFilePaths.length
      ? minify(versionedBundle, {
          mangle: false,
          keep_classnames: true,
        }).then(({ code }) => {
          assert.string(code);
          console.log(`Minified bundle size: ${code.length} bytes`);
          return terseFilePaths.map(dest => write(dest, code));
        })
      : []
    ),
```